### PR TITLE
fix(client): allow regex to match windows-like paths

### LINKED ISSUE #42719: Error when changing markdown files in curriculum/ on Windows
When, on Windows, changing markdown files in curriculum/ they don't get reloaded because of error occurring. 

```js
TypeError: Cannot read property '1' of null
    at replaceChallengeNode (*\freeCodeCamp\client\utils\buildChallenges.js:20:50)
    at FSWatcher.<anonymous> (*\freeCodeCamp\client\plugins\fcc-source-challenges\gatsby-node.js:41:9)
    at FSWatcher.emit (events.js:315:20)
    at FSWatcher.emitWithAll (*\freeCodeCamp\client\node_modules\chokidar\index.js:540:8)
    at FSWatcher._emit (*\freeCodeCamp\client\node_modules\chokidar\index.js:632:8)
    at listener (*\freeCodeCamp\client\node_modules\chokidar\lib\nodefs-handler.js:389:18)
    at *\freeCodeCamp\client\node_modules\chokidar\lib\nodefs-handler.js:271:49
    at foreach (*\freeCodeCamp\client\node_modules\chokidar\lib\nodefs-handler.js:41:5)
    at StatWatcher.<anonymous> (*\freeCodeCamp\client\node_modules\chokidar\lib\nodefs-handler.js:271:11)
    at StatWatcher.emit (events.js:315:20)
    at StatWatcher.onchange (internal/fs/watchers.js:63:8)
```

This is caused by the windows-like paths, that don't get matched at https://github.com/freeCodeCamp/freeCodeCamp/blob/71a7ab5c0f0a6dd97d8c5a087b67c30c0572c70a/client/utils/buildChallenges.js#L19

**Screenshots**
image

**System (please complete the following information):**
 - OS: Windows

### PR DESCRIPTION
**Pull request by gikf**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #42719

### CODE REVIEW

File: client/utils/buildChallenges.js [hunk 1]
```diff
@@ -16,8 +16,8 @@ exports.localeChallengesRootDir = getChallengesDirForLang(curriculumLocale);
 exports.replaceChallengeNode = () => {
   return async function replaceChallengeNode(filePath) {
     // get the meta so that challengeOrder is accurate
-    const blockNameRe = /\d\d-[-\w]+\/([^/]+)\//;
-    const blockName = filePath.match(blockNameRe)[1];
+    const blockNameRe = /\d\d-[-\w]+([\/\\])([^/]+)\1/;
+    const blockName = filePath.match(blockNameRe)[2];
```
Comment (ShaunSHamilton) [inline client/utils/buildChallenges.js:20]:
```suggestion
    const blockNameRe = /\d\d-[-\w]+\/([^/]+)\//;
    const posix = path
      .normalize(filePath)
      .split(path.sep)
      .join(path.posix.sep);
    const blockName = posix.match(blockNameRe)[1];
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
What about something we have done in the past: https://github.com/freeCodeCamp/freeCodeCamp/pull/39454/files

File: client/utils/buildChallenges.js (hunk 1 above)
Comment (gikf) [reply client/utils/buildChallenges.js:20]: Good idea :rocket:

**Review by ShaunSHamilton — APPROVED** [review]
This appears to work 👍

**Comment by Sembauke** [comment]
Hey @ShaunSHamilton,

I do not know on which OS you develop, but it would be handy to have someone test this on Linux and Windows.

**Comment by ShaunSHamilton** [comment]
> Hey @ShaunSHamilton,
> 
> I do not know on which OS you develop, but it would be handy to have someone test this on Linux and Windows.

I have tested on Windows 10 and Ubuntu 20.04 (WSL 2)

**Review by raisedadead — APPROVED** [review]

**Pull request merged**